displayServer.go

```diff
@@ -1,46 +1,161 @@
 package main
 
 import (
+	"bytes"
+	"context"
+	"embed"
+	"encoding/json"
 	"fmt"
-	"log"
+	"io/fs"
 	"net"
 	"net/http"
 	"os"
 	"os/signal"
+	"strings"
 	"time"
 
 	"github.com/google/uuid"
+	"github.com/gorilla/mux"
 	"github.com/koron/go-ssdp"
+	rt "github.com/wailsapp/wails/v2/pkg/runtime"
 )
 
-/*
-	This is very _very_ temporary and just for proof of concept.
-	Ideally we expose this in some way that actually interacts with the rest of the app.
+//go:embed frontend-display
+var displayServerFrontend embed.FS
 
-	This is mainly to test the SSDP functionality and make sure it works.
-*/
+type DisplayServerData struct {
+	Type string
+	Meta map[string]interface{}
+}
+
+type DisplayServer struct {
+	ctx      context.Context
+	server   *http.Server
+	data     DisplayServerData
+	dataChan chan DisplayServerData
+}
+
+func NewDisplayServer() *DisplayServer {
+	return &DisplayServer{
+		dataChan: make(chan DisplayServerData),
+	}
+}
 
-func startDisplayServer() {
+func (d *DisplayServer) SetData(data DisplayServerData) {
+	rt.LogDebugf(d.ctx, "DisplayServer.SetData: %s", data)
+	d.data = data
+	d.dataChan <- data
+}
 
-	go ssdpInit()
+func (d *DisplayServer) startup(ctx context.Context) {
+	d.ctx = ctx
 
-	fmt.Println("Starting display server")
+	go d.ssdpInit()
 
-	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprintf(w, "Simple Worship Display")
-	})
+	router := mux.NewRouter()
 
-	err := http.ListenAndServe(":7777", nil)
+	var frontendFS = fs.FS(displayServerFrontend)
+	staticContent, err := fs.Sub(frontendFS, "frontend-display/src")
 	if err != nil {
-		log.Fatal("ListenAndServe: ", err)
+		rt.LogErrorf(d.ctx, "Error: %e", err)
 	}
 
+	httpfs := http.FileServer(http.FS(staticContent))
+	router.PathPrefix("/src/").Handler(http.StripPrefix("/src/", httpfs))
+
+	router.HandleFunc("/data", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
This will serve a json representation of the current data (used when the page loads for the first time)